jaxtyping/_import_hook.py

```diff
@@ -385,6 +428,7 @@ def f(x: Float32[Array, "batch channels"]):
     else:
         raise RuntimeError("Cannot find a PathFinder in sys.meta_path")
 
+    _update_typechecker_decorator(typechecker)
```

Comment:
And add a warning to switch from the string-based approach to the function based approach?

Reply:
Are you sure about this one? It seems quite hard to pass a function instead of a string, if you wanted to pass a string, so if somebody passes a function probably they know what they are doing. And then they would need to disable the warning, since it might annoy them..

I reckon that the function-style typechecker should be used only by advanced users, since in normal applications you never want to pass a function, so you point does make sense. But in any case, showing a warning if you do use such functionality seems a bit weird to me.

Reply:
Ah, so my thinking was that we'd make the function-passing form the recommended approach. I think it's a lot less magical to do so! Eval'ing code is pretty unpleasant business. The recommended approach becomes:
```python
import beartype
import jaxtyping
with jaxtyping.install_import_hook(..., beartype.beartype):
    ...
```

What's your thinking for the opposite?

Reply:
Oh, no. I just completely missed the possibility to the thing you propose, because I thought that this just won't work.. But it does certainly work (unless users break beartype in the hooked file, but I think that is extremely unlikely).